contrib/platform/test/com/sun/jna/platform/win32/Win32ServiceDemo.java

```diff
@@ -251,10 +251,15 @@ public void init() {
      *
      * @param machine name of the machine or null for localhost
      * @param access access flags
-     * @return handle to ServiceControlManager or null when failed
+     * @return handle to ServiceControlManager
+     * @throws Win32Exception If manager not opened
      */
     private static SC_HANDLE openServiceControlManager(String machine, int access) {
-        return Advapi32.INSTANCE.OpenSCManager(machine, null, access);
+        SC_HANDLE handle = Advapi32.INSTANCE.OpenSCManager(machine, null, access);
+        if (handle == null) {
+            throw new Win32Exception(Kernel32.INSTANCE.GetLastError());
```

Comment:
The places this method is called presently perform a null-check that is made unnecessary by this change.  Since you're manually throwing the exception here, I don't see why you can't just keep this method as-is, returning null, and add `assertNotNull()` after the calling function, processing the `GetLastError` there.

Reply:
I *could* call GetLastError outside, but then what's the point of the function?
If the caller needs to know what the function does, shouldn't the entire content of the function be copied to all those places?

What I want to avoid is have test cases failing with "N/A" or "null" as the reason as it's not helping. If the error is simply "Access denied", it makes more sense and there is less need to dig into why the test case failed.

Reply:
I see your point and I'm fine with the exception here, but I still think you need to go to the places where this is called, and look at the logic there.  It may be some resources need to be closed in a finally block depending on which call throws the exception.

Reply:
I have checked all 4 places, and the only thing I saw was that if there were an exception inside the if-statement for the manager, then the manager was not closed.
I don't think this is so serious since this is only test framework for JNA, but I switched to try-finally to ensure that manager is closed if acquired.

Reply:
You can remove the null checks. :)

Reply:
Did you check the change or just commented on my comment?
I did remove the if-null check for the manager, wasn't that your point?

Reply:
Sorry again. I saw "open" and "servicemanager" on the same line and missed that it's a different method call. At this point I think it's fine. Waiting to merge until after "code freeze". :)

Reply:
No worries.
How long is the freeze? Is there any date set to the release of 5.6?

Reply:
This coming week.